plugin-elasticsearch/src/api/ser.rs

```diff
@@ -0,0 +1,157 @@
+//! Implementation of a small subset of the REST API of OpenSearch/ElasticSearch.
+
+use alumet::{
+    measurement::{MeasurementBuffer, MeasurementPoint, WrappedMeasurementValue},
+    pipeline::elements::output::OutputContext,
+};
+
+use serde::{ser::Error, ser::SerializeMap, Serialize};
+use serde_json::json;
+use std::{collections::HashMap, time::SystemTime};
+use time::{format_description::well_known::Rfc3339, UtcDateTime};
+
+/// OpenSearch/ElasticSearch serializer helper.
+pub struct Serializer {
+    /// Each index will be named like `"{index_prefix}-{metric_name}"`
+    pub index_prefix: String,
+
+    pub metric_unit_as_index_suffix: bool,
+}
+
+impl Serializer {
+    /// Generates the mappings for an index.
+    pub fn common_index_mappings(&self) -> serde_json::Value {
+        fn field_with_type(field: &str, data_type: &str) -> (String, serde_json::Value) {
+            (
+                field.to_string(),
+                json!({
+                    "type": data_type
+                }),
+            )
+        }
+
+        let index_props = serde_json::Map::from_iter([
+            field_with_type("@timestamp", "date_nanos"),
```

Comment:
maybe this behavior of defining fields and types should be placed as a `DocMeasurement` method, as it's his own specific knowledge (note that `DocMeasurement` mentions about the columns names in the `serialize` method).